pyro/poutine/runtime.py

```diff
@@ -196,3 +151,52 @@ def am_i_wrapped():
     :returns: bool
     """
     return len(_PYRO_STACK) > 0
+
+
+def effectful(fn=None, type=None):
```

Comment:
Can we memoize this function to reduce computational overhead of abstraction?

Reply:
Hmm, memoization seems to be causing a memory leak in `ss_vae_M2.py`.

Reply:
Ok, we can avoid memoizing. I guess that would be premature optimization anyway.